libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/BotFrameworkOptions.cs

```diff
@@ -8,17 +8,36 @@
 
 namespace Microsoft.Bot.Builder.Integration.AspNet.Core
 {
+    /// <summary>
+    /// Contains settings that your ASP.NET application uses to initialize the <see cref="BotAdapter"/>
+    /// that it adds to the HTTP request pipeline.
+    /// </summary>
+    /// <seealso cref="ApplicationBuilderExtensions"/>
     public class BotFrameworkOptions
     {
         private readonly List<IMiddleware> _middleware;
 
+        /// <summary>
+        /// Creates a <see cref="BotFrameworkOptions"/> object.
+        /// </summary>
         public BotFrameworkOptions()
         {
             _middleware = new List<IMiddleware>();
         }
 
+        /// <summary>
+        /// The credential provider with which to initialize the adapter.
+        /// </summary>
         public ICredentialProvider CredentialProvider { get; set; }
+
+        /// <summary>
+        /// The middleware collection with which to initialize the adapter.
+        /// </summary>
         public IList<IMiddleware> Middleware { get => _middleware; }
+
+        /// <summary>
+        /// Indicates whether to enable the proactive messages endpoint for the bot.
```

Comment:
Are you able to add signifigantly more to this one? This is a *HUGE* can of worms, and better docs here would be excellent.

Reply:
@cleemullins , I'm vague on this one. I'll ask for more info from @drub0y , and then flesh this out.